ote/test/clj/ote/transit_changes/detection_test.clj

```diff
@@ -169,10 +173,302 @@
+        route-name-2 ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]} ;; first current week (11.2.)
+         {route-name ["h1" "!!" "!!" "h4" "h5" "h6" "h7"]
+        route-name-2 ["h1" "!!" "!!" "h4" "h5" "h6" "h7"]} ;; 18.2. first change -> only one found currently | needs to detect change also on 13.2. -> this is set to current week
+         {route-name ["h1" "!!" "!!" "h4" "h5" "h6" "h7"]
+        route-name-2 ["h1" "!!" "!!" "h4" "h5" "h6" "h7"]} ;; no changes here (25.2.)
+         {route-name ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]
+        route-name-2 ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]} ;; Only tuesday is found (4.3.)
+         {route-name ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]
+        route-name-2 ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]} ;;
+         {route-name ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]
+        route-name-2 ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]} ;;
+         {route-name ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]
+        route-name-2 ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]}))
+
+(def data-two-week-change
+  (weeks (d 2019 2 4)
+         {route-name ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]} ;; 4.2.
+         {route-name ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]} ;; first current week (11.2.)
+         {route-name ["h1" "!!" "!!" "h4" "h5" "h6" "h7"]}
+         {route-name ["h1" "!!" "!!" "h4" "h5" "h6" "h7"]}
+         {route-name ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]}
+         {route-name ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]}
+         {route-name ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]}
+         {route-name ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]}))
+
+(deftest more-than-one-change-found
+  (spec-test/instrument `detection/route-weeks-with-first-difference)
+
+  ;; first test that the test data and old change detection code agree
+  (testing "single-change detection code agrees with test data"
+    (is (= (d 2019 2 18) (-> test-more-than-one-change
+                             detection/route-weeks-with-first-difference
+                             (get route-name)
+                             :different-week
+                             :beginning-of-week))))
+  
+  (let [diff-pairs (-> test-more-than-one-change
+                       detection/changes-by-week->changes-by-route
+                       detection/detect-changes-for-all-routes)
+        ;; diff-pairs (detection/routes-changed-weeks test-more-than-one-change)
+        old-diff-pairs (detection/route-weeks-with-first-difference test-more-than-one-change)]
+    (testing "got two changes"
+      (is (= 2 (count diff-pairs))))
+    (testing "first change is detected"
+      (is (= (d 2019 2 18) (-> diff-pairs
+                               first
+                               :different-week
+                               :beginning-of-week))))
+
+    (testing "second change is detected"
+      (is (= (d 2019 3 4) (-> diff-pairs
+                              second
+                              :different-week
+                              :beginning-of-week))))))
+
+
```

Comment:
there are now several defined test datas that are not unused, do we need to add more tests to use them or should unused data be removed?